### PR TITLE
[LOOP-4953] Favorite Food Insights

### PR DESCRIPTION
**Pull request by SwiftlyNoah**
[JIRA](https://tidepool.atlassian.net/jira/software/c/projects/LOOP/issues/LOOP-4956?jql=project%20%3D%20%22LOOP%22%20ORDER%20BY%20created%20DESC) |  Goes with [Loop PR](https://github.com/tidepool-org/Loop/pull/691)

- New `GlucoseCarbChart` shows glucose data with carb icons representing carb entries along the x-axis
- Carb icons show whether or not a food is a favorite food
- When carb entry is highlighted, highlight layer is temporarily tinted carb-green rather than its default glucose tint color.

Glucose+Carb
HighlightedGlucose+Carb
Glucose+HighlightedCarb

### CODE REVIEW

File: LoopKit/CarbKit/CarbStore.swift [hunk 1]
```diff
@@ -282,9 +284,10 @@ extension CarbStore {
     ///   - start: The earliest date of values to retrieve
     ///   - end: The latest date of values to retrieve, if provided
     ///   - dateAscending: Indicates how the result will be sorted. If true, function returns `[oldestEntry, ..., newestEntry]`. If false, function returns`[newestEntry, ..., oldestEntry]`
+    ///   - fetchLimit: Limit on the maximum number of entries to be retrieved
     ///   - with: The favorite food identifier to match
     /// - Returns: An array of carb entries, sorted by startDate, or error
-    private func getCarbEntries(start: Date? = nil, end: Date? = nil, dateAscending: Bool = true, with favoriteFoodID: String? = nil ) -> Result<[StoredCarbEntry], Error> {
+    private func getCarbEntries(start: Date? = nil, end: Date? = nil, dateAscending: Bool = true, fetchLimit: Int? = nil, with favoriteFoodID: String? = nil) -> Result<[StoredCarbEntry], Error> {
```
Comment (ps2) [inline LoopKit/CarbKit/CarbStore.swift:290]: I don't see fetchLimit being used here?

File: LoopKitUI/Extensions/ChartPoint.swift [hunk 2]
```diff
@@ -127,6 +127,53 @@ extension ChartPoint: TimelineValue {
     }
 }
 
+extension Array where Element: ChartPoint {
+    /// Merges the current sorted array with another sorted array into a single sorted array.
```
Comment (ps2) [inline LoopKitUI/Extensions/ChartPoint.swift]: Maybe just mention what the sort key is (startDate?), so that they know which sorting is being maintained.

**Review by ps2 — APPROVED** [review]
Minor nits, but otherwise LGTM!

File: LoopKit/CarbKit/CarbStore.swift (hunk 1 above)
Comment (SwiftlyNoah) [reply LoopKit/CarbKit/CarbStore.swift:290]: Good catch, thanks

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**